jax/_src/numpy/reductions.py

```diff
@@ -2482,12 +2490,85 @@ def _quantile(a: Array, q: Array, axis: int | tuple[int, ...] | None,
     axis = canonicalize_axis(axis, a.ndim)
 
+  q, = promote_dtypes_inexact(q)
+  q = jnp.atleast_1d(q)
   q_shape = q.shape
   q_ndim = q.ndim
   if q_ndim > 1:
     raise ValueError(f"q must be have rank <= 1, got shape {q.shape}")
 
   a_shape = a.shape
+  # Handle weights
+  if weights is None:
+    a, = promote_dtypes_inexact(a)
+  else:
+    a, weights = promote_dtypes_inexact(a, weights)
+    a_shape = a.shape
+    w_shape = np.shape(weights)
+    if w_shape != a_shape:
+      if axis is None:
+        raise TypeError("Axis must be specified when shapes of a and weights differ.")
+      if isinstance(axis, tuple):
+        if w_shape != tuple(a_shape[i] for i in axis):
+            raise ValueError("Shape of weights must match the shape of the axes being reduced.")
+        weights = lax.broadcast_in_dim(
+            weights,
+            shape=a_shape,
+            broadcast_dimensions=axis
+        )
+    else:
+        if len(w_shape) != 1 or w_shape[0] != a_shape[axis]:
+            raise ValueError("Length of weights not compatible with specified axis.")
+        weights = lax.expand_dims(weights, axis)
+        weights = _broadcast_to(weights, a.shape)
+     
+
+    if squash_nans:
+      nan_mask = ~lax_internal._isnan(a)
+      weights = _where(nan_mask, weights, 0)
+    else:
+      with jax.debug_nans(False):
+        a = _where(any(lax_internal._isnan(a), axis=axis, keepdims=True), np.nan, a)
+
+    total_weight = sum(weights, axis=axis, keepdims=True)
+    a_sorted, weights_sorted = lax.sort_key_val(a, weights, dimension=axis)
+    cum_weights = lax.cumsum(weights_sorted, axis=axis)
+    cum_weights_norm = lax.div(cum_weights, total_weight)
+
+    def _weighted_quantile(qi):
+      index_dtype = dtypes.default_int_dtype()
+      idx = sum(lax.lt(cum_weights_norm, qi), axis=axis, dtype=index_dtype, keepdims=keepdims)
+      idx = lax.clamp(0, idx, a_sorted.shape[axis] - 1)
+      val = jnp.take_along_axis(a_sorted, idx, axis)
+
+      idx_prev = lax.clamp(idx - 1, 0, a_sorted.shape[axis] - 1)
+      val_prev = jnp.take_along_axis(a_sorted, idx_prev, axis)
+      cw_prev = jnp.take_along_axis(cum_weights_norm, idx_prev, axis)
+      cw_next = jnp.take_along_axis(cum_weights_norm, idx, axis)
+
+      if method == "linear":
+        denom = cw_next - cw_prev
+        denom = _where(denom == 0, 1, denom)
+        weight = (qi - cw_prev) / denom
+        out = val_prev * (1 - weight) + val * weight
+      elif method == "lower":
+        out = val_prev
+      elif method == "higher":
+        out = val
+      elif method == "nearest":
+        out = _where(lax.abs(qi - cw_prev) < lax.abs(qi - cw_next), val_prev, val)
+      elif method == "midpoint":
+        out = (val_prev + val) / 2
+      elif method == "inverted_cdf":
+        out = val
+      else:
+        raise ValueError(f"{method=!r} not recognized")
+      return out
+    
+    result = jax.vmap(_weighted_quantile)(q)
+    if q.shape == (1,):
+        result = result[0]
```

Comment:
This will return in incorrect shape in the case that the user passes `q` with shape `(1,)`.